pkg/logs/input/container/launcher.go

```diff
@@ -58,7 +59,13 @@ func NewLauncher(collectAll bool, collectFromFiles bool, sources *config.LogSour
 		log.Infof("Could not setup the kubernetes launcher: %v", err)
 	}
 
-	log.Infof("Container logs won't be collected")
+	launcher, err = docker.NewLauncher(sources, services, pipelineProvider, registry, true)
+	if err != nil {
+		log.Warnf("Could not setup the docker launcher: %v. Will not be able to collect container logs", err)
+		return nil
```

Comment:
@L3n41c I don't think this code path can ever be executed at the moment given the logic in `docker.NewLauncher`, but the caller of `NewLauncher` does not filter out `nil` as a return value and will try to dereference the return value.

(you'll note that this method previously always returned a non-nil value, either an actual launcher or a `NoopLauncher`).

Reply:
I think we can prove that `docker.NewLauncher(…)` cannot return an error when called with `shouldRetry = true` but it is definitively safer for the future to make this more robust.
I’ll return a `NewNoopLauncher()` instead of `nil`.

Reply:
Here is the change: https://github.com/DataDog/datadog-agent/pull/4839/commits/e067865926497120f140da2763c3b8ee1489d650#diff-4787c98f3435a93c3f818cf181cc10f2